controllers/datadogagent/override/podtemplatespec_test.go

```diff
@@ -41,9 +40,9 @@ func TestPodTemplateSpec(t *testing.T) {
 			},
 		},
 		{
-			name: "override image name and tag",
+			name: "01: given URI, override image name, tag, JMX",
```

Comment:
nit: I'm not sure about numbered tests. When tests are updated you need to make sure that everything is still sorted. In my experience this can result in missing numbers, unsorted lists, etc.

Reply:
I totally agree that this requires renumbering and did it couple of times, but with over dozen somewhat similar tests it made debugging and navigating in the IDE easier and found the additional effort worth it. For most other tests numbering is likely overkill but for this it was worth additional effort so I'd rather keep it.

Reply:
:+1: